### PR TITLE
Create page improvements

### PR DESCRIPTION
**Pull request by davidpede**
### This pr resolves the following
- Fixes user unable to create pages unless they have fred permission 'new_document_in_root'
- Adds parent default selection functionality

### Why is this needed

- User unable to create pages unless they have fred permission 'new_document_in_root'

Currently if a user has `new_document` permission but not `new_document_in_root` then they cannot create documents in front end. The 'Create Page' btn in tree is not rendered and the 'Create Child Page' btn is broken as `buildCreatePage()` is not called so `this.parentInput.choices` is never populated (https://github.com/modxcms/fred/blob/master/_build/assets/js/Components/Sidebar/Pages/Pages.js#L50-L52).

Also enabling the 'Create Page' btn with only `new_document` permission seems better for the user as more intuitive than clicking on tree and seeing if they can add a child page.

- Add parent default selection functionality

Changes the default selection for the 'Parent' drop down from 'No Parent' to the following (in order):
1. Current Resource Parent - If it **IS NOT** root
2. Current Resource - If it **IS** root

'No Parent' is still an option in the parent select, but only if the user has `new_document_in_root` permissions.

### CODE REVIEW

**Comment by davidpede** [comment]
Needs a bit more work this - left a commit out so page won't save :weary: I'll add to pr when back in office

File: _build/assets/js/Components/Sidebar/Pages/Pages.js [hunk 1]
```diff
@@ -127,7 +130,11 @@ export default class Pages extends Sidebar {
                 choices : this.parents,
                 shouldSort: false
             }
-        }, this.state.parent, onChangeChoices);
+        }, this.state.parent, onChangeChoices, (setting, label, select, choicesInstance, defaultValue) => {
+            let id = fredConfig._resource.parent ? fredConfig._resource.parent : fredConfig._resource.id;
```
[the remark this answers is not in the document]
Comment (theboxer) [reply _build/assets/js/Components/Sidebar/Pages/Pages.js]: Also the original line 194 with `this.createPageButton = dt('fred.fe.pages.create_page', ['fred--accordion-plus'], this.openCreatePage);` Right after this line, there should be 
```js
if (!fredConfig.permission.new_document_in_root) {
    this.createPageButton.setAttribute('hidden', 'hidden');
}
```

**Comment by davidpede** [comment]
I've made the changes you requested except:
> Also the original line 194 with `this.createPageButton = dt('fred.fe.pages.create_page', ['fred--accordion-plus'], this.openCreatePage);` Right after this line, there should be
> 
> ```js
> if (!fredConfig.permission.new_document_in_root) {
>     this.createPageButton.setAttribute('hidden', 'hidden');
> }
> ```
I see what your saying however could you reconsider from a content creator pov? I've got an early adopter group of 20+ that have mostly all brought this up as an issue for them.

They have naturally adopted the approach of navigating to their relevant sections of a website in frontend, then creating pages from there. Non of them can create in root, just within certain sections of the site, up to 5 levels deep (from root). It annoys them then having to re-navigate the tree every time to create a child page.

They want to just click 'Site > Create Page', have the parent pre-selected to where they are. (They can change it if necessary).

If they have permission then they can still select 'No Parent' and create in root but this is so rare as normally a sites top level is fixed by an admin or the like and rarely changed. The content is created deeper in the tree.

What'd you think?

**Comment by theboxer** [comment]
How about:
```js
if (!fredConfig.permission.new_document_in_root && !fredConfig._resource.parent) {
    this.createPageButton.setAttribute('hidden', 'hidden');
}
```

**Comment by davidpede** [comment]
Yep that works for me, thanks for the suggestion

**Comment by theboxer** [comment]
Awesome, thank you for your work on this.

**Pull request merged**